app/src/main/java/com/amaze/filemanager/activities/MainActivity.java

```diff
@@ -612,18 +616,65 @@ private void initializeInteractiveShell() {
         }
     }
 
+    /**
+     * @return paths to all available volumes in the system (include emulated)
+     */
+    public synchronized ArrayList<StorageDirectoryParcelable> getStorageDirectories() {
+        ArrayList<StorageDirectoryParcelable> volumes;
+        if (SDK_INT >= Build.VERSION_CODES.N) {
+            volumes = getStorageDirectoriesNew();
+        } else {
+            volumes = getStorageDirectoriesLegacy();
+        }
+        if (isRootExplorer()) {
+            volumes.add(new StorageDirectoryParcelable("/", getResources().getString(R.string.root_directory), R.drawable.ic_drawer_root_white));
+        }
+        return volumes;
+    }
+
+    /**
+     * @return All available storage volumes (including internal storage, SD-Cards and USB devices)
+     */
+    @TargetApi(Build.VERSION_CODES.N)
+    public synchronized ArrayList<StorageDirectoryParcelable> getStorageDirectoriesNew() {
+        // Final set of paths
+        ArrayList<StorageDirectoryParcelable> volumes = new ArrayList<>();
+        StorageManager sm = getSystemService(StorageManager.class);
+        for (StorageVolume volume : sm.getStorageVolumes()) {
+            Log.i("MainActivity", "Volume: " + volume + "");
```

Comment:
Please delete this log call. (If you allow contributor changes I can push changes directly)

Reply:
I won't be able to do it today, but AFAICT contributor changes are enabled.

Thanks!